### PR TITLE
[ALLUXIO-3040] Update scripts and docs to remove per-compute build profiles

### PR DESCRIPTION
**Pull request by apc999**
https://alluxio.atlassian.net/browse/ALLUXIO-3040

### CODE REVIEW

File: core/client/runtime/pom.xml [hunk 1]
```diff
@@ -162,6 +162,23 @@
           </execution>
         </executions>
       </plugin>
+      <plugin>
+        <groupId>com.coderplus.maven.plugins</groupId>
+        <artifactId>copy-rename-maven-plugin</artifactId>
```
Comment (calvinjia) [inline core/client/runtime/pom.xml]: Would it be possible to use one of our existing plugins?

File: dev/scripts/generate-tarball.sh [hunk 2]
```diff
@@ -61,36 +56,13 @@ function prepare_repo {
...
-  if [[ "${build_all_client_profiles}" == true ]]; then
-    for profile in "${EXTRA_SYMLINK_CLIENT_PROFILES[@]}"; do
-      mkdir -p "${CLIENT_DIR}/${profile}"
-      pushd "${CLIENT_DIR}/${profile}" >> /dev/null
-      ln -s "../default/alluxio-${version}-default-client.jar" "alluxio-${version}-${profile}-client.jar"
-      popd >> /dev/null
-    done
+  local hadoop_profile=$1
+  echo "Running build and logging to ${BUILD_LOG}"
+  local profiles_arg="-Pmesos"
```
Comment (calvinjia) [inline dev/scripts/generate-tarball.sh]: Unrelated to this PR, should we remove the mesos profile and make it built by default?

File: docs/_config.yml [hunk 3]
```diff
@@ -14,12 +14,7 @@ ALLUXIO_MASTER_VERSION_SHORT: master
 ALLUXIO_MAJOR_VERSION: master
 # We must inline the version string (e.g., "1.4.0-SNAPSHOT") rather than using the macro of Alluxio version.
 # Otherwise the macro name remains in the output.
-ALLUXIO_CLIENT_JAR_PATH: /<PATH_TO_ALLUXIO>/client/default/alluxio-1.7.0-SNAPSHOT-default-client.jar
-ALLUXIO_CLIENT_JAR_PATH_BUILD: /<PATH_TO_ALLUXIO>/core/client/runtime/target/alluxio-core-client-runtime-1.7.0-SNAPSHOT-jar-with-dependencies.jar
-ALLUXIO_CLIENT_JAR_PATH_FLINK: /<PATH_TO_ALLUXIO>/client/flink/alluxio-1.7.0-SNAPSHOT-flink-client.jar
-ALLUXIO_CLIENT_JAR_PATH_HADOOP: /<PATH_TO_ALLUXIO>/client/hadoop/alluxio-1.7.0-SNAPSHOT-hadoop-client.jar
-ALLUXIO_CLIENT_JAR_PATH_PRESTO: /<PATH_TO_ALLUXIO>/client/presto/alluxio-1.7.0-SNAPSHOT-presto-client.jar
-ALLUXIO_CLIENT_JAR_PATH_SPARK: /<PATH_TO_ALLUXIO>/client/spark/alluxio-1.7.0-SNAPSHOT-spark-client.jar
+ALLUXIO_CLIENT_JAR_PATH: /<PATH_TO_ALLUXIO>/client/alluxio-1.7.0-SNAPSHOT-client.jar
```
Comment (calvinjia) [inline docs/_config.yml]: Do we still want to keep this path, or just direct to the `core/client/runtime` path?

File: core/client/runtime/pom.xml (hunk 1 above)
Comment (apc999) [reply core/client/runtime/pom.xml]: discussed offline. the current resources plugin can not rename, and we want to keep the convention that in `target` dir, uber jars are named after `*-jar-with-dependencies.jar`

File: dev/scripts/generate-tarball.sh (hunk 2 above)
Comment (apc999) [reply dev/scripts/generate-tarball.sh]: I am cool with this proposal, though I don't have too much context on the mesos integration. @aaudiber or @madanadit may have a better idea.

File: docs/_config.yml (hunk 3 above)
Comment (apc999) [reply docs/_config.yml]: discussed offline. `core/client/runtime/target` is too deep and not proper to stay in the final release tarball. I will keep the top level `client/` dir and add it to `.gitignore`

**Comment by apc999** [comment]
@calvinjia  PTAL

File: dev/scripts/generate-tarball.sh (hunk 2 above)
Comment (aaudiber) [reply dev/scripts/generate-tarball.sh]: +1 for removing the profile

**Pull request merged**